dowhy/gcm/whatif.py

```diff
@@ -167,3 +168,72 @@ def _evaluate_intervention(node: Any,
+    is do(T := 0), then the average causal effect is given by ACE = E[Y | do(T := 1)] - E[Y | do(T := 0)]:
+        >>> average_causal_effect(causal_model, 'Y', {'T': lambda _ : 1}, {'T': lambda _ : 0})
+
+    We can also specify more complex interventions on multiple nodes:
+        >>> average_causal_effect(causal_model,
+        >>>                       'Y',
+        >>>                       {'T': lambda _ : 1, 'X0': lambda x : x + 1},
+        >>>                       {'T': lambda _ : 0, 'X0': lambda x : x * 2})
+    In the above, we would estimate ACE = E[Y | do(T := 1), do(X0 := X0 + 1)] - E[Y | do(T := 0), do(X0 := X0 * 2)].
+
+    Note: Only continuous real-valued and binary targets are supported currently.
+
+    :param causal_model: The probabilistic causal model we perform this intervention on .
+    :param target_node: Target node for which the ACE is estimated.
+    :param interventions_alternative: Dictionary defining the interventions for the alternative values.
+    :param interventions_reference: Dictionary defining the interventions for the reference values.
+    :param observed_data: Factual data that we observe for the nodes in the causal graph. By default, new data
+                          is sampled using the causal model. If observational data is available, providing them
+                          might improve the accuracy by mitigating issues due to a misspecified graph and/or causal
+                          models.
+    :param num_samples_to_draw: Number of samples drawn from the causal model for estimating ACE if no observed data is
+                                given.
+    :return: The estimated average causal effect (ACE).
+    """
+    causal_model = ProbabilisticCausalModel(node_connected_subgraph_view(causal_model.graph, target_node))
+
+    validate_causal_dag(causal_model.graph)
+    for node in interventions_alternative:
+        validate_node_in_graph(causal_model.graph, node)
+    for node in interventions_reference:
+        validate_node_in_graph(causal_model.graph, node)
+
+    if observed_data is None and num_samples_to_draw is None:
+        raise ValueError("Either observed_samples or num_samples_to_draw need to be set!")
+    if observed_data is not None and num_samples_to_draw is not None:
+        raise ValueError("Either observed_samples or num_samples_to_draw need to be set, not both!")
+
+    if num_samples_to_draw is not None:
+        observed_data = draw_samples(causal_model, num_samples_to_draw)
+
+    samples_from_target_alt = _interventional_samples(causal_model, observed_data, interventions_alternative)[
+        target_node].to_numpy()
+    samples_from_target_ref = _interventional_samples(causal_model, observed_data, interventions_reference)[
+        target_node].to_numpy()
+
+    target_causal_model = causal_model.causal_mechanism(target_node)
+    if isinstance(target_causal_model, ClassifierFCM):
```

Comment:
this may be the right place to add inline comment saying that discrete variables are not supported.

Reply:
Only categorical target nodes with more than 2 categories are not supported. The input can be categorical, discrete or continuous and the target can be discrete, continuous or categorical with 2 categories (i.e. binary).

Reply:
Added inline comment.